src/app/application.cpp

```diff
@@ -907,20 +907,32 @@ int Application::exec()
             QCoreApplication::exit(EXIT_FAILURE);
         connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(EXIT_FAILURE); });
 
+        printf("%s", qUtf8Printable(u"\n******** %1 ********\n"_s.arg(tr("Information"))));
+
         const Preferences *pref = Preferences::instance();
 
-        const auto scheme = pref->isWebUiHttpsEnabled() ? u"https"_s : u"http"_s;
-        const auto url = u"%1://localhost:%2\n"_s.arg(scheme, QString::number(pref->getWebUiPort()));
-        const QString mesg = u"\n******** %1 ********\n"_s.arg(tr("Information"))
-                + tr("To control qBittorrent, access the WebUI at: %1").arg(url);
-        printf("%s\n", qUtf8Printable(mesg));
+        if (pref->isWebUiEnabled())
+        {
+            const auto scheme = pref->isWebUiHttpsEnabled() ? u"https"_s : u"http"_s;
+            auto serverAddress = pref->getWebUiAddress();
+
+            if ((serverAddress == u"*") || serverAddress.isEmpty())
```

Comment:
Discussion: should it be generalized with

https://github.com/qbittorrent/qBittorrent/blob/24fbaa40b7256f2d252217ae3267e3571ac0b473/src/webui/webui.cpp#L71-L72

i.e. extracted to a separate function to avoid duplicated logic?

Reply:
IMO, the better way is to provide such things via `WebUI` class interface, e.g. `QHostAddress WebUI::serverAddress() const`.

Reply:
I also thought about that. In fact, I think ideally this part should not read preferences at all, but get all information from the class.

Reply:
> this part should not read preferences at all, but get all information from the class.

:+1:

Reply:
Done.